cmd/controller-manager/app/helper_test.go

```diff
@@ -60,6 +60,20 @@ func TestIsControllerEnabled(t *testing.T) {
 			disabledByDefaultControllers: []string{"delta", "echo"},
 			expected:                     false,
 		},
+		{
+			name:                         "on by all except",
+			controllerName:               "alpha",
+			controllers:                  []string{"*", "-charlie"},
+			disabledByDefaultControllers: []string{"delta", "echo"},
+			expected:                     true,
+		},
+		{
+			name:                         "off by all except",
```

Comment:
name seems confusing for the case.
The case seems to be you have set it default on and then explicitly turned off the relevant controller.
How about something like "on by default, off by name"?
I like the test case, it makes the precedence order very clear.

Reply:
Done. I changed to `on by star, not off by name` and `off by name with star`